iwyu_include_picker.cc

```diff
@@ -1387,14 +1400,21 @@ vector<string> IncludePicker::GetCandidateHeadersForFilepathIncludedFrom(
   // ConvertToQuotedInclude because it avoids trouble when the same
   // file is accessible via different include search-paths, or is
   // accessed via a symlink.
+  // We also try to recalculate any quoted includes using absolute paths,
+  // because they can likely be converted to an include relative to the
+  // current file.
   vector<string> retval;
   for (MappedInclude& mapped_include : mapped_includes) {
     const string& quoted_include_as_written =
       MaybeGetIncludeNameAsWritten(including_filepath, mapped_include.path);
-    if (quoted_include_as_written.empty()) {
-      retval.push_back(mapped_include.quoted_include);
-    } else {
+    if (!quoted_include_as_written.empty()) {
       retval.push_back(quoted_include_as_written);
+    } else if (StartsWith(mapped_include.quoted_include, "\"/") &&
```

Comment:
Is this a check for whether the path is absolute? It probably won't work for Windows paths. I wonder if you should instead strip quotes/angles and use `IsAbsolutePath`...

Reply:
Done, via a new method on `MappedInclude`.